### PR TITLE
[probably not ready to merge] Add experimental graph.py type annotations

### PR DESCRIPTION
**Pull request by eshansingh**
This is my best guess on the types. They're still returning errors, but only a few, and most everything else is OK.

### CODE REVIEW

**Comment by drvinceknight** [comment]
Could you add this file to https://github.com/Axelrod-Python/Axelrod/blob/master/run_mypy.py, this will ensure that the continuous integration (the automatic testing) also checks the types.

File: axelrod/graph.py [hunk 1]
```diff
@@ -6,7 +6,11 @@
 """
 
 from collections import defaultdict
+from typing import List, Dict, Hashable, Tuple
+from numbers import Number
 
+Vertice = Hashable
```
Comment (marcharper) [inline axelrod/graph.py]: Vertice -> Vertex (and below)

File: axelrod/graph.py [hunk 2]
```diff
@@ -6,7 +6,11 @@
 """
 
 from collections import defaultdict
+from typing import List, Dict, Hashable, Tuple
+from numbers import Number
```
Comment (marcharper) [inline axelrod/graph.py]: We can assume that the weight is a float IMO rather than a Number

Comment (eshansingh) [reply axelrod/graph.py]: Sure, will make changes ASAP.

**Pull request closed without merging**